can't build resource files on linux because of SpecStrings.h

Resource compiler started working; new issue is only on Linux host builds. This commit re-enables Versioning on MacOS https://github.com/farzonl/warfLang/pull/28/commits/a0c28ce1667e6a796d1b2435b30ec7bf4421a58a. This change makes the Disable version exclusive to linux to windows cross builds https://github.com/farzonl/warfLang/pull/28/commits/b919eb168f9b1a20a581f4ec5f3fcf7e65430a78.

The issue seems to be with the vfs_overlay.yaml file. That means Resource files faile to buld for the same reason we can't cross build the unit tests on Linux (https://github.com/farzonl/warfLang/issues/29).

The error in the logs:
```
022-10-03T19:49:38.2208981Z /home/runner/work/warfLang/warfLang/cmake_config/../packages/win_sysroot/WINSDK/10.0.22621.0/include/um/winver.h:18:10: fatal error: 'SpecStrings.h' file not found
2022-10-03T19:49:38.2209584Z #include <SpecStrings.h>
2022-10-03T19:49:38.2210055Z          ^~~~~~~~~~~~~~~
2022-10-03T19:49:38.2210584Z 1 error generated.
```

All the below cmake and code changes have been removed. A simpler check was added to `src/cli/CMakeLists.txt`
```
if(WIN32 AND NOT DISABLE_VERSION)
list(APPEND SOURCE_FILES
    ${CMAKE_CURRENT_BINARY_DIR}/../lib/Version/VersionResource.rc
    ${CMAKE_CURRENT_BINARY_DIR}/../lib/Version/VersionInfo.h
)
endif()
```

~~On both Mac and Linux I had to disable the invocation of the resouce compilerhttps://github.com/farzonl/warfLang/pull/28~~

~~That was the reason for the cmake changes  that looked like~~
```
if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR 
   CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND NOT CMAKE_CROSS_COMPILING)

...

endif()
```

~~and the code changes that look like~~
```cpp
#if !defined(DISABLE_VERSION)
...
#endif
```

new issue

```
022-10-03T19:49:38.2208981Z /home/runner/work/warfLang/warfLang/cmake_config/../packages/win_sysroot/WINSDK/10.0.22621.0/include/um/winver.h:18:10: fatal error: 'SpecStrings.h' file not found
2022-10-03T19:49:38.2209584Z #include <SpecStrings.h>
2022-10-03T19:49:38.2210055Z          ^~~~~~~~~~~~~~~
2022-10-03T19:49:38.2210584Z 1 error generated.
```

--- a/.github/workflows/cmake-win-llvm-cross.yml
+++ b/.github/workflows/cmake-win-llvm-cross.yml
@@ -88,7 +88,6 @@ jobs:
       if: matrix.os == 'ubuntu-latest'
       run: |
        cmake -GNinja -S . -B build  \
-       -DDISABLE_VERSION=TRUE \
        -DBuildTest=TRUE \
        -DCMAKE_TOOLCHAIN_FILE=cmake_config/llvm-win-cross.cmake
 

--- a/Dockerfile.llvm-win-cross
+++ b/Dockerfile.llvm-win-cross
@@ -38,16 +38,16 @@ COPY cmake_config cmake_config
 RUN cmake -GNinja -S . -B winBuild  \
     -DCMAKE_BUILD_TYPE=Release \
     -DBuildTest=TRUE \
-    -DDISABLE_VERSION=TRUE \
     -DCMAKE_TOOLCHAIN_FILE=cmake_config/llvm-win-cross.cmake
 
 RUN ninja -C./winBuild
-#
-## To get wine to run 32 and 64 bit binaries you have to do the install in stages
-##RUN dpkg --add-architecture i386
-##RUN apt-get update \
-##    && apt install -y \
-##    wine32
+
+# To get wine to run 32 and 64 bit binaries you have to do the install in stages
+#RUN dpkg --add-architecture i386
+#RUN apt-get update \
+#    && apt install -y \
+#    wine32
+
 ENV WINEPATH="/root/project/packages/win_sysroot/MSVC/14.11.25547/lib/x64"
 ENV WINEPATH="$WINEPATH;/root/project/packages/win_sysroot/WINSDK/10.0.22621.0/lib/um/x64"
 ENV WINEPATH="$WINEPATH;/root/project/packages/win_sysroot/WINSDK/10.0.22621.0/lib/ucrt/x64"

--- a/cmake_config/llvm-win-cross.cmake
+++ b/cmake_config/llvm-win-cross.cmake
@@ -76,3 +76,5 @@ set(F_FLAGS "-fms-compatibility-version=19.11 -fms-extensions -fdelayed-template
 set(COMPILE_FLAGS "--target=${TARGET_TRIPLE} -nostdlib -lmsvcrt ${F_FLAGS}" )
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}  ${COMPILE_FLAG}" CACHE STRING "" FORCE)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILE_FLAG}" CACHE STRING "" FORCE)
+
+set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} -Xclang -ivfsoverlay -Xclang ${WiN_VFS_OVERLAY}")

--- a/cmake_config/llvm-win-cross.cmake
+++ b/cmake_config/llvm-win-cross.cmake
@@ -76,5 +76,4 @@ set(F_FLAGS "-fms-compatibility-version=19.11 -fms-extensions -fdelayed-template
 set(COMPILE_FLAGS "--target=${TARGET_TRIPLE} -nostdlib -lmsvcrt ${F_FLAGS}" )
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}  ${COMPILE_FLAG}" CACHE STRING "" FORCE)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILE_FLAG}" CACHE STRING "" FORCE)
-
 set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} -Xclang -ivfsoverlay -Xclang ${WiN_VFS_OVERLAY}")

--- a/src/cli/CMakeLists.txt
+++ b/src/cli/CMakeLists.txt
@@ -4,7 +4,7 @@ set(SOURCE_FILES
                  # add files here
                  main.cpp)
 
-if(WIN32 AND NOT DISABLE_VERSION)
+if(WIN32)
 list(APPEND SOURCE_FILES
     ${CMAKE_CURRENT_BINARY_DIR}/../lib/Version/VersionResource.rc
     ${CMAKE_CURRENT_BINARY_DIR}/../lib/Version/VersionInfo.h

